src/documentCreator/services/token.ts

```diff
@@ -0,0 +1,64 @@
+import { TokenRegistry, WriteableTitleEscrowOwner, TitleEscrowOwner } from "@govtechsg/oa-token";
+import { WrappedDocument } from "@govtechsg/open-attestation";
+import { getData } from "@govtechsg/tradetrust-schema";
+import { getLogger } from "../logger";
+import { Wallet, ethers } from "ethers";
+import { get } from "lodash";
+
+const { trace } = getLogger("saga:tokenService");
+
+interface InitializeTokenInterface {
+  document: WrappedDocument<any>;
+  web3Provider: ethers.providers.BaseProvider | undefined;
+  wallet: Wallet | undefined;
+}
+
+let registryInstance: TokenRegistry;
+let titleEscrowOwnerInstance: TitleEscrowOwner;
+
+export const initializeTokenInstance = async ({
+  document,
+  web3Provider = undefined,
+  wallet = undefined
+}: InitializeTokenInterface): Promise<void> => {
+  trace(`web3 provider is: ${web3Provider} and wallet is: ${wallet}`);
+  trace(`document to initialize ${JSON.stringify(document)}`);
+  if (!document || !web3Provider || !wallet) throw new Error("Can not initialize the token instance");
+  const documentData = getData(document);
+  const contractAddress = get(documentData, "issuers[0].tokenRegistry", "");
```

Comment:
Are we going to throw when there is more than one issuer? I thought we handled this somewhere in oa-token? Would we want to use that function from there instead?